### PR TITLE
Add smart_proxy_shellhooks (RPM)

### PR DESCRIPTION
**Pull request by lzap**
For plugin updates, please indicate which repos this should be built into:

* Nightly
* 2.3

Do not merge yet, need to test the build.

### CODE REVIEW

**Comment by evgeni** [comment]
```
error: Installed (but unpackaged) file(s) found:
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/README
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/ansible-tower-cli
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_args
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_body
```

**Comment by evgeni** [comment]
```
error: Installed (but unpackaged) file(s) found:
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/README
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/ansible-tower-cli
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_args
   /usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/examples/print_body
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/README
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/ansible-tower-cli
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/print_args
   /var/lib/rubygem-smart_proxy_shellhooks/shellhooks/print_body
```

`/var/lib` sounds wrong to me?

**Comment by lzap** [comment]
Do not merge, still testing.

**Comment by lzap** [comment]
Now it finally works:

```[root@katello ~]# rpm -ql tfm-rubygem-smart_proxy_shellhooks
/etc/foreman-proxy/settings.d/shellhooks.yml
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/LICENSE
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/shellhooks.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/shellhooks_api.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/shellhooks_http_config.ru
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_shellhooks-0.9.0/lib/smart_proxy_shellhooks/version.rb
/opt/theforeman/tfm/root/usr/share/gems/specifications/smart_proxy_shellhooks-0.9.0.gemspec
/usr/share/foreman-proxy/bundler.d/shellhooks.rb
/var/lib/foreman-proxy/shellhooks
/var/lib/foreman-proxy/shellhooks/README
/var/lib/foreman-proxy/shellhooks/ansible-tower-cli
/var/lib/foreman-proxy/shellhooks/print_args
/var/lib/foreman-proxy/shellhooks/print_body

[root@katello ~]# curl -k https://localhost:9090/shellhook/
{"valid":["ansible-tower-cli","print_args","print_body"],"invalid":["README"]}
```

**Comment by lzap** [comment]
I am gonna push 0.9.1 out and rebuild once again.

**Comment by lzap** [comment]
It works fine now. RTM

**Comment by lzap** [comment]
Hello, anyone there? I will be working on Debian package this week.

File: packages/plugins/rubygem-smart_proxy_shellhooks/rubygem-smart_proxy_shellhooks.spec [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-proxy-plugin-%{plugin_name} = %{version}
+# end specfile generated dependencies
+
+%{?scl:Obsoletes: rubygem-%{gem_name}}
```
Comment (evgeni) [inline packages/plugins/rubygem-smart_proxy_shellhooks/rubygem-smart_proxy_shellhooks.spec:41]: you can drop that, we never had a rubygm-shellhooks outside the scl.

**Comment by evgeni** [comment]
[test rpm]

While not packaging related, I find the "invalid: README" weird -- should that be special-handled in the plugin?

**Review by evgeni — APPROVED** [review]
the `Obsoletes` is not needed, but also doesn't hurt.

ACK, please ping if you want to fix the obsoletes.

File: packages/plugins/rubygem-smart_proxy_shellhooks/rubygem-smart_proxy_shellhooks.spec (hunk 1 above)
Comment (lzap) [reply packages/plugins/rubygem-smart_proxy_shellhooks/rubygem-smart_proxy_shellhooks.spec:41]: Dropped

**Comment by lzap** [comment]
I dropped the line, rebased. Thanks.

> While not packaging related, I find the "invalid: README" weird -- should that be special-handled in the plugin?

Since `README` is not executable, it's reported as an invalid shell hook. Not sure what exactly do you find weird. This endpoint is never called like that, this is more like a debugging endpoint. Foreman only does `POST` here.

**Comment by evgeni** [comment]
you don't seem to actually have pushed the changes :)

**Pull request merged**